### ISSUE TITLE
Investigate performance of current classification method

### ISSUE THREAD

**Issue by LoyVanBeek** [tue-robotics/tue_robocup#511]
(i.e. inception_v3 retrained on RoboCup data)

- [x] Find the images logged on RoboCup Japan
- [x] Annotate those and put them in the corresponding folders
- [x] Run the evaluate-classifier script with the network we trained at RoboCup Japan on those images. (~/MEGA/data/rwc2017/models/tensorflow_ros/output_graph.pb)

**Comment by LoyVanBeek** [comment]
@Rayman Where are the logged images of RoboCup Japan? In MEGA/data/rwc2017, there's only model and training data but no images logged during challenges.

**Comment by Rayman** [comment]
`AMIGO{1,2}:~/MEGA/...`

**Comment by LoyVanBeek** [comment]
OK, but that's not synced to MEGA so I can have the data on my laptop as well?

**Comment by MatthijsBurgh** [comment]
On Friday I let AMIGO sync to MEGA. Although I had to leave before it was finished, so not all data is online.

**Comment by LoyVanBeek** [comment]
Can you start a sync it again maybe tomorrow morning?

**Comment by MatthijsBurgh** [comment]
I will have time at 11:00. So I will power-up AMIGO at 11.

**Comment by LoyVanBeek** [comment]
The collected data is at ~/MEGA/data/rwc2017/training_data/annotated/verified/.
Analysis is running :-)

**Comment by LoyVanBeek** [comment]
Results of rwc2017:

![robocup_rwc2017_result](https://user-images.githubusercontent.com/709259/33316495-637ac98e-d434-11e7-80ac-395a4f665612.png)

**Comment by LoyVanBeek** [comment]
Conclusions based on this graph: 

- Not very bad, with the exception  of the classes Curry and Cup-star
- Curry and Cup-star are often confused. These classes do look very very similar indeed. 
![Cup star](https://user-images.githubusercontent.com/709259/33526305-4a9665cc-d83f-11e7-8e4f-a3917c991b71.jpg) ![Curry](https://user-images.githubusercontent.com/709259/33526306-4df8b512-d83f-11e7-810b-93e80bfb6c2d.jpg) The only difference is in the color of the lid.
- A decision threshold of ca. 0.15 should have weeded out most false positives

Recommendations:

1. Pay more attention to color differences? 
2. In cases where 2 classes are both quite likely, perhaps take a **simple** other method to decide (of course feeding the network more data to also make this difference should also work). This can decide on a very obvious feature like color. The difference between Cup-star and Curry is color so perhaps a color histogram trained on only those classes could help. But then we need tools to point out where to focus on.

**Comment by LoyVanBeek** [comment]
@Rayman, do you know on how many pictures this classifier was trained? The logging acquired some 7500 images over the complete tournament, but the training was done on less, I assume.

**Comment by Rayman** [comment]
At least 20 pictures per class. I don't know exactly. Probably half of the total images.

**Comment by reinzor** [comment]
But now the question. What is the accuracy if we choose the best threshold?

**Comment by LoyVanBeek** [comment]
For RoboCup, we're judged on the amount of true positives (+10) and false positives (-5). The fraction of those is in the graphs. 

It should be possible to (automatically) determine an optimal threshold, based on hypothetically showing one instance of each class to the robot, for example.

**Comment by LoyVanBeek** [comment]
But, as mentioned in https://github.com/tue-robotics/tue_robocup/issues/511#issuecomment-347444533, this is tested on the training data (although I **assumed** the model was not trained on the full raining data we collected during challenges) and this is not as representative as I'd hope. 

InceptionV3 on ca 50 images for each of 23 classes, the performance is down to 0.65 accuracy, which is quite low. Capturing more images during RoboCup is too time consuming, so that is our constraint.
